frontend/lib/search/search_page.dart

```diff
@@ -50,14 +68,22 @@ class _SearchPageState extends State<SearchPage> {
             ResultsLoader(
               searchText: searchFieldText,
               categoryIds: _selectedCategories.map((e) => e.id).toList(),
-              coordinates: currentCoordinatesInput,
+              coordinates: _sortingMode == SortingMode.byDistance ? _coordinates : null,
             )
           ],
         ),
         LocationButton(
```

Comment:
We have two `LocationButtons` (map/search) i think since we now switch different sortings with this button we should rename it to "SortingButton" or sth. similar.